spec/system/admin/order_spec.rb

```diff
@@ -956,42 +956,60 @@ def new_order_with_distribution(distributor, order_cycle)
+              click_link 'Ship Order'
+
+              within ".reveal-modal" do
+                expect(page).to have_checked_field('Send a shipment/pick up ' \
+                                                   'notification email to the customer.')
+                expect {
+                  find_button("Confirm").click
+                }.to enqueue_job(ActionMailer::MailDeliveryJob).exactly(:once)
+              end
 
-            within ".reveal-modal" do
-              expect(page).to have_checked_field('Send a shipment/pick up ' \
-                                                 'notification email to the customer.')
-              expect {
-                find_button("Confirm").click
-              }.to enqueue_job(ActionMailer::MailDeliveryJob).exactly(:once)
+              expect(order.reload.shipped?).to be true
+              expect(page).to have_text 'SHIPPED'
             end
 
-            expect(order.reload.shipped?).to be true
-            expect(page).to have_text 'SHIPPED'
-          end
+            it "ships the order without sending email" do
+              click_on subpage
+              expect(order.reload.shipped?).to be false
 
-          it "ships the order without sending email" do
-            expect(order.reload.shipped?).to be false
+              find('.ofn-drop-down').click
+              click_link 'Ship Order'
 
-            find('.ofn-drop-down').click
-            click_link 'Ship Order'
+              within ".reveal-modal" do
+                uncheck 'Send a shipment/pick up notification email to the customer.'
+                expect {
+                  find_button("Confirm").click
+                }.not_to enqueue_job(ActionMailer::MailDeliveryJob)
+              end
 
-            within ".reveal-modal" do
-              uncheck 'Send a shipment/pick up notification email to the customer.'
-              expect {
-                find_button("Confirm").click
-              }.not_to enqueue_job(ActionMailer::MailDeliveryJob)
+              expect(order.reload.shipped?).to be true
+              expect(page).to have_text 'SHIPPED'
             end
-
-            expect(order.reload.shipped?).to be true
-            expect(page).to have_text 'SHIPPED'
           end
         end
+
+        it_behaves_like "ship order from dropdown", "Order Details"
+        it_behaves_like "ship order from dropdown", "Customer Details", "Payments",
+                        "Adjustments", "Invoices", "Return Authorizations" do
+          before { pending("#12369") }
+        end
+        context "" do
+          before { pending("#12369") }
+          it_behaves_like "ship order from dropdown", "Customer Details"
+          it_behaves_like "ship order from dropdown", "Payments"
+          it_behaves_like "ship order from dropdown", "Adjustments"
+          it_behaves_like "ship order from dropdown", "Invoices"
+          it_behaves_like "ship order from dropdown", "Return Authorizations"
+        end
```

Comment:
Isn't this a duplicate of the line above ?

Reply:
> Isn't this a duplicate of the line above ?

Apologies @rioug , my mistake... I'm not sure both notations work. I think we need a separate line, for each value of |subpage| so I found this (the current) code to work best.